components/web_discovery/browser/web_discovery_service.cc

```diff
@@ -104,6 +120,35 @@ void WebDiscoveryService::OnPatternsLoaded() {
     content_scraper_ =
         std::make_unique<ContentScraper>(server_config_loader_.get());
   }
+  if (!double_fetcher_) {
+    double_fetcher_ = std::make_unique<DoubleFetcher>(
+        profile_prefs_.get(), shared_url_loader_factory_.get(),
+        base::BindRepeating(&WebDiscoveryService::OnDoubleFetched,
+                            base::Unretained(this)));
+  }
+  if (!reporter_) {
+    reporter_ = std::make_unique<Reporter>(
+        profile_prefs_.get(), shared_url_loader_factory_.get(),
+        credential_manager_.get(), server_config_loader_.get());
+  }
+  MaybeSendAliveMessage();
+}
+
+void WebDiscoveryService::OnDoubleFetched(
+    const GURL& url,
+    const base::Value& associated_data,
+    std::optional<std::string> response_body) {
+  if (!response_body) {
+    return;
+  }
+  auto prev_scrape_result = PageScrapeResult::FromValue(associated_data);
+  if (!prev_scrape_result) {
+    return;
+  }
+  content_scraper_->ParseAndScrapePage(
```

Comment:
this function looks pretty heavy, why don't we do it on the task runner?

Reply:
we do, inside the method itself